src/main/java/org/opentripplanner/middleware/auth/Auth0Connection.java

```diff
@@ -151,4 +157,38 @@ private static JWTVerifier getVerifier(Request req, String token) {
     public static boolean authDisabled() {
         return hasConfigProperty("DISABLE_AUTH") && "true".equals(getConfigPropertyAsText("DISABLE_AUTH"));
     }
+
+    /**
+     * Confirm that the user exists
+     */
+    private static Auth0UserProfile isValidUser(Request request) {
+
+        Auth0UserProfile profile = getUserFromRequest(request);
+        if (profile == null || (profile.adminUser == null  && profile.otpUser == null && profile.apiUser == null)) {
+            logMessageAndHalt(request, HttpStatus.NOT_FOUND_404, "Unknown user.");
+        }
+
+        return profile;
+    }
+
+    /**
+     * Confirm that the user's actions are on their items if not admin.
+     */
+    public static void isAuthorized(String userId, Request request) {
+
+        Auth0UserProfile profile = isValidUser(request);
+
+        // let admin do anything
+        if (profile.adminUser != null) {
+            return;
+        }
+
+        if (userId == null ||
+            (profile.otpUser != null && !profile.otpUser.id.equalsIgnoreCase(userId)) ||
```

Comment:
Here also, this should be `.equals(...)`

Reply:
yeah, will be in the next commit.